File: Mozu.SiteBuilder.UX.Admin/Api/Models/Shipping/SiteShippingOriginAddress.cs
using System;
using System.Runtime.Serialization;

namespace Mozu.SiteBuilder.UX.Admin.Api.Models.Shipping
{
    [DataContract]
    public class SiteShippingOriginAddress
    {
        [DataMember(Name = "id")]
        public int? Id { get; set; }

        [DataMember(Name = "senderName")]
        public string SenderName { get; set; }

        [DataMember(Name = "address1")]
        public string Address1 { get; set; }

        [DataMember(Name = "address2")]
        public string Address2 { get; set; }

        [DataMember(Name = "address3")]
        public string Address3 { get; set; }

        [DataMember(Name = "cityOrTown")]
        public string CityOrTown { get; set; }

        [DataMember(Name = "stateOrProvince")]
        public string StateOrProvince { get; set; }

        [DataMember(Name = "country")]
        public string Country { get; set; }

        [DataMember(Name = "postalOrZipCode")]
        public string PostalOrZipCode { get; set; }

        [DataMember(Name = "createBy")]
        public string CreateBy { get; set; }

        [DataMember(Name = "createDate")]
        public DateTime? CreateDate { get; set; }

        [DataMember(Name = "updateBy")]
        public string UpdateBy { get; set; }

        [DataMember(Name = "updateDate")]
        public DateTime? UpdateDate { get; set; }
    }
}

File: Mozu.SiteBuilder.UX.Admin/Scripts/app/model/SiteShippingOriginAddress.js
/**
 * @class Taco.model.SiteShippingOriginAddress
 */
Ext.define('Taco.model.SiteShippingOriginAddress', {
    extend: 'Taco.core.data.Model',
    fields: [
        { name: 'senderName', type: 'string' },
        { name: 'address1', type: 'string' },
        { name: 'address2', type: 'string' },
        { name: 'address3', type: 'string' },
        { name: 'cityOrTown', type: 'string' },
        { name: 'stateOrProvince', type: 'string' },
        { name: 'country', type: 'string' },
        { name: 'postalOrZipCode', type: 'string' }
    ],

    proxy: {
        type: 'ajaxproxy',
        api: {
            read: '/admin/app/shipping/originaddress/read',
            create: '/admin/app/shipping/originaddress/create',
            update: '/admin/app/shipping/originaddress/edit',
            destroy: '/admin/app/shipping/originaddress/delete'
        },
        reader: {
            type: 'json',
            root: 'items',
            successProperty: 'success',
            messageProperty: "message"
        },
        writer: {
            allowSingle: true,
            type: 'json'
        }
    }
});